tests/js/blockPRNG.scrypttest.js

```diff
@@ -0,0 +1,185 @@
+const { expect } = require('chai');
+const { bsv, buildContractClass, PubKey, toHex, Sha256, Bytes, getPreimage, signTx, buildTypeClasses, num2bin } = require('scryptlib');
+const { num2hex, sha256d, reverseEndian, compileContract, inputIndex, inputSatoshis, uint322bin, toLittleIndian, pdiff2Target } = require('../../helper');
+const BN = bsv.crypto.BN
+
+function newTx() {
+    const utxo = {
+        txId: 'c1d32f28baa27a376ba977f6a8de6ce0a87041157cef0274b20bfda2b0d8df96',
+        outputIndex: 0,
+        script: '',   // placeholder
+        satoshis: inputSatoshis
+    };
+    return new bsv.Transaction().from(utxo);
+}
+
+
+let merklePath = {
+    "blockHash": "0000000000000000091216c46973d82db057a6f9911352892b7769ed517681c3",
+    "branches": [
+        {
+            "hash": "7e0ba1980522125f1f40d19a249ab3ae036001b991776813d25aebe08e8b8a50",
+            "pos": "R"
+        },
+        {
+            "hash": "1e3a5a8946e0caf07006f6c4f76773d7e474d4f240a276844f866bd09820adb3",
+            "pos": "R"
+        }
+    ],
+    "hash": "c1d32f28baa27a376ba977f6a8de6ce0a87041157cef0274b20bfda2b0d8df96",
+    "merkleRoot": "95a920b1002bed05379a0d2650bb13eb216138f28ee80172f4cf21048528dc60"
+}
+
+let wrongMerklePath = {
+    "blockHash": "0000000000000000091216c46973d82db057a6f9911352892b7769ed517681c3",
+    "branches": [
+        {
+            "hash": "7e0ba1980522125f1f40d19a249ab3ae036001b991776813d25aebe08e8b8a50",
+            "pos": "R"
+        },
+        {
+            "hash": "1e3a5a8946e0caf07006f6c4f76773d7e474d4f240a276844f866bd09820adb3",
+            "pos": "L"
+        }
+    ],
+    "hash": "c1d32f28baa27a376ba977f6a8de6ce0a87041157cef0274b20bfda2b0d8df96",
+    "merkleRoot": "95a920b1002bed05379a0d2650bb13eb216138f28ee80172f4cf21048528dc60"
+}
+
+
+// block hash: 0000000000000000091216c46973d82db057a6f9911352892b7769ed517681c3
+// height: 575191
+let header = {
+    version: 536870912,
+    previousblockhash: "00000000000000000012ce7f4bbce3346ac438ab7fdcb6fa5440db9857856a7f",
+    merkleroot: "95a920b1002bed05379a0d2650bb13eb216138f28ee80172f4cf21048528dc60",
+    time: 1553501874,
+    bits: "180978d5",
+    nonce: 2482491775,
+    difficulty: 116078424449.9863
+}
+
```

Comment:
Similar code in other scrypttest files of this PR. Put in a shared file. Do not duplicate large chunks of code.